### ISSUE TITLE
[Accessibility]: Tree page UI (header, buttons, history) does not reflow

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35239]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34420

### Problem description

**Issue 1**
Repo header is unusable:
<img width="365" alt="image" src="https://user-images.githubusercontent.com/9516420/167642112-e7bb3d3c-4741-47a1-bd96-9fc31c5d22a8.png">

**Issue 2**
Tabs are difficult to read
<img width="447" alt="image" src="https://user-images.githubusercontent.com/9516420/167642186-8c89aaea-4398-4029-8afb-dfaae189efca.png">

**Issue 3**
Commit messages are unreadable
<img width="254" alt="image" src="https://user-images.githubusercontent.com/9516420/167642230-b987edc3-e642-437c-ba63-b62e21cf4c66.png">

### Expected behavior

**Issue 1**
We should try to break this out onto new lines if possible. Note: this could be quite complex, we should break this out into a different issue if required

**Issue 2**
We can use `flex-direction: column` here to just render the tabs vertically. That should improve UX for mobile significantly.

**Issue 3**
We should break the message out onto new lines, rather than truncate it here on mobile

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
